test/unit/tab/index.test.tsx

```diff
@@ -35,143 +37,153 @@ test('adds a class from state.classList', () => {
 });
 
 test('#adapter.addClass adds class to state.classList', () => {
-  const wrapper = shallow(<Tab />);
+  const wrapper = shallow<Tab>(<Tab />);
   wrapper.instance().adapter.addClass('test-class');
   assert.isTrue(wrapper.state().classList.has('test-class'));
 });
 
 test('#adapter.removeClass removes class from state.classList', () => {
-  const wrapper = shallow(<Tab />);
+  const wrapper = shallow<Tab>(<Tab />);
   wrapper.setState({classList: new Set(['test-class'])});
   wrapper.instance().adapter.removeClass('test-class');
   assert.isFalse(wrapper.state().classList.has('test-class'));
 });
 
 test('#adapter.hasClass returns true if state.classList contains class', () => {
-  const wrapper = shallow(<Tab />);
+  const wrapper = shallow<Tab>(<Tab />);
   wrapper.setState({classList: new Set(['test-class'])});
   const hasClass = wrapper.instance().adapter.hasClass('test-class');
   assert.isTrue(hasClass);
 });
 
 test('#adapter.hasClass returns false if state.classList contains class', () => {
-  const wrapper = shallow(<Tab />);
+  const wrapper = shallow<Tab>(<Tab />);
   const hasClass = wrapper.instance().adapter.hasClass('test-class');
   assert.isFalse(hasClass);
 });
 
 test('#adapter.setAttr sets tabIndex on state', () => {
-  const wrapper = shallow(<Tab />);
-  wrapper.instance().adapter.setAttr('tabIndex', 0);
+  const wrapper = shallow<Tab>(<Tab />);
+  wrapper.instance().adapter.setAttr('tabIndex', '0');
   assert.equal(wrapper.state().tabIndex, 0);
 });
 
 test('#adapter.setAttr sets aria-selected on state', () => {
-  const wrapper = shallow(<Tab />);
+  const wrapper = shallow<Tab>(<Tab />);
   wrapper.instance().adapter.setAttr('aria-selected', true);
   assert.isTrue(wrapper.state()['aria-selected']);
 });
 
 test('#adapter.getOffsetLeft returns tabElement_.offsetLeft', () => {
-  const wrapper = mount(<Tab />);
-  assert.equal(wrapper.instance().adapter.getOffsetLeft(), wrapper.instance().tabElement_.current.offsetLeft);
+  const wrapper = mount<Tab>(<Tab />);
+  assert.equal(
+    wrapper.instance().adapter.getOffsetLeft(),
+    wrapper.instance().tabElement_.current!.offsetLeft
+  );
 });
 
 test('#adapter.getOffsetWidth returns tabElement_.offsetWidth', () => {
-  const wrapper = mount(<Tab>Text</Tab>);
-  assert.equal(wrapper.instance().adapter.getOffsetWidth(), wrapper.instance().tabElement_.current.offsetWidth);
+  const wrapper = mount<Tab>(<Tab>Text</Tab>);
+  assert.equal(
+    wrapper.instance().adapter.getOffsetWidth(),
+    wrapper.instance().tabElement_.current!.offsetWidth
+  );
 });
 
 test('#adapter.getContentOffsetLeft returns tabContentElement_.offsetLeft', () => {
-  const wrapper = mount(<Tab>Text</Tab>);
+  const wrapper = mount<Tab>(<Tab>Text</Tab>);
   assert.equal(
     wrapper.instance().adapter.getContentOffsetLeft(),
-    wrapper.instance().tabContentElement_.current.offsetLeft);
+    wrapper.instance().tabContentElement_.current!.offsetLeft
```

Comment:
What is the exclamation mark doing here?

Reply:
It thinks it is possible that current might be null. So I just used the `!` to tell its wrong